caddyhttp/httpserver/plugin.go

```diff
@@ -239,9 +239,18 @@ func (h *httpContext) MakeServers() ([]caddy.Server, error) {
 		// see if all the addresses (both sites and
 		// listeners) are loopback to help us determine
 		// if this is a "production" instance or not
+		for _, listenHost := range cfg.ListenHosts {
+			if !atLeastOneSiteLooksLikeProduction {
+				if !caddy.IsLoopback(cfg.Addr.Host) &&
+					!caddy.IsLoopback(listenHost) &&
+					(caddytls.QualifiesForManagedTLS(cfg) ||
+						caddytls.HostQualifies(cfg.Addr.Host)) {
+					atLeastOneSiteLooksLikeProduction = true
+				}
+			}
+		}
 		if !atLeastOneSiteLooksLikeProduction {
 			if !caddy.IsLoopback(cfg.Addr.Host) &&
-				!caddy.IsLoopback(cfg.ListenHost) &&
```

Comment:
Can you explain this change and the added lines above this?

Reply:
lines added: loop over cfg.ListenHosts, same logic like the old.
line removed: if cfg.ListenHosts is empty, just don't check it.

Reply:
seems i forget to ensure the second block only runs when cfg.ListenHosts is empty